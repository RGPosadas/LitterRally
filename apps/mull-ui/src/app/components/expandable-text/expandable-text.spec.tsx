import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

import ExpandableText from './expandable-text';

describe('ExpandableText', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<ExpandableText>Text</ExpandableText>);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<ExpandableText>Text</ExpandableText>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show all text and hide button if cutoff is too big', () => {
    const dom = render(<ExpandableText cutoff={100}>Text</ExpandableText>);

    const div = dom.getByTestId('expandable-text-div');

    expect(div.textContent).toBe('Text');
    const button = dom.queryByTestId('expandable-text-button');

    expect(button).toBe(null);
  });

  it('should cutoff text by default', () => {
    const dom = render(<ExpandableText cutoff={2}>Text</ExpandableText>);

    const div = dom.getByTestId('expandable-text-div');

    expect(div.textContent).toBe('Te... more');
  });

  it('should show all text on press', () => {
    const dom = render(<ExpandableText cutoff={2}>Text</ExpandableText>);

    const div = dom.getByTestId('expandable-text-div');
    const button = dom.getByTestId('expandable-text-button');

    ReactTestUtils.Simulate.click(button);

    expect(div.textContent).toBe('Text less');
  });
});
