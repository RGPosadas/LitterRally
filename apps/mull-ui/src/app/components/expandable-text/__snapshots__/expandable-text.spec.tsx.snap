// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandableText should match snapshot 1`] = `
<div
  className=""
  data-testid="expandable-text-div"
>
  Text
</div>
`;
